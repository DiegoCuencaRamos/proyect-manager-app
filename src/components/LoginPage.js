import React from 'react'
import { useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import { 
    startGoogleLogin, 
    startFacebookLogin, 
    startTwitterLogin, 
    startGithubLogin 
} from '../actions/auth'

export const LoginPage = () => {
    // 1. Dispatch
    const dispatch = useDispatch()

    // 2. Events
    const onGoogleLogin = () => {
        dispatch(startGoogleLogin())
    }

    const onFacebookLogin = () => {
        dispatch(startFacebookLogin())
    }

    const onTwitterLogin = () => {
        dispatch(startTwitterLogin())
    }

    const onGithubLogin = () => {
        dispatch(startGithubLogin())
    }

    // 3. Render
    return (
        <section className="login-layout">
            <div className="login-layout__box">
                <h1 className="login-layout__title">Proyect Manager</h1>
                <p>It is time to get control over your proyects.</p>
                <div
                    className="btn--google"
                    onClick={onGoogleLogin}
                >
                    <i className="fab fa-google"></i>
                    <span>Sign in with Google</span>
                </div>
                <div
                    className="btn--facebook"
                    onClick={onFacebookLogin}
                >
                    <i className="fab fa-facebook-f"></i>
                    <span>Sign in with Facebook</span>
                </div>
                <div
                    className="btn--twitter"
                    onClick={onTwitterLogin}
                >
                    <i className="fab fa-twitter"></i>
                    <span>Sign in with Twitter</span>
                </div>
                <div
                    className="btn--github"
                    onClick={onGithubLogin}
                >
                    <i className="fab fa-github"></i>
                    <span>Sign in with Github</span>
                </div>
                <Link
                    className="btn--emial"
                    to='/email-login'
                >
                    <i className="fas fa-envelope"></i>
                    <span>Sign in with Email</span>
                </Link>
            </div>
        </section>
)}

export default LoginPage