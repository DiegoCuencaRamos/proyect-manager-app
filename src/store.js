import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store