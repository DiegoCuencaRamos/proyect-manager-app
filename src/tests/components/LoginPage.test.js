import React from 'react'
import { mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import store from '../../store'
import LoginPage from '../../components/LoginPage'

const getWrapper = () => mount(
    <MemoryRouter>
        <Provider store={store}>
            <LoginPage />
        </Provider>
    </MemoryRouter>
)

test('Should render LoginPage correctly', () => {
    const wrapper = render(
        <MemoryRouter>
            <Provider store={store}>
                <LoginPage />
            </Provider>
        </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
})
