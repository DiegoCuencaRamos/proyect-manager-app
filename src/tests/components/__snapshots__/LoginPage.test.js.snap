// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage correctly 1`] = `
<section
  class="login-layout"
>
  <div
    class="login-layout__box"
  >
    <h1
      class="login-layout__title"
    >
      Proyect Manager
    </h1>
    <p>
      It is time to get control over your proyects.
    </p>
    <div
      class="btn--google"
    >
      <i
        class="fab fa-google"
      />
      <span>
        Sign in with Google
      </span>
    </div>
    <div
      class="btn--facebook"
    >
      <i
        class="fab fa-facebook-f"
      />
      <span>
        Sign in with Facebook
      </span>
    </div>
    <div
      class="btn--twitter"
    >
      <i
        class="fab fa-twitter"
      />
      <span>
        Sign in with Twitter
      </span>
    </div>
    <div
      class="btn--github"
    >
      <i
        class="fab fa-github"
      />
      <span>
        Sign in with Github
      </span>
    </div>
    <a
      class="btn--emial"
      href="/email-login"
    >
      <i
        class="fas fa-envelope"
      />
      <span>
        Sign in with Email
      </span>
    </a>
  </div>
</section>
`;
